server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/StarTreeTestUtils.java

```diff
@@ -201,11 +202,13 @@ public static void validateFileFormats(
             if (rootNode.getChildren() != null) {
                 sortedChildren = new ArrayList<>(rootNode.getChildren().values());
             }
-
-            if (starTreeNode.getChildDimensionId() != -1) {
+            int dimensionId = starTreeNode.getChildDimensionId();
```

Comment:
Can you assert the newly added metadata as part of assertStarTreeMetadata ? we need to make sure both docValuesType and comparatorType of dimension config are asserted

we can make this check to explicitly validate the above 2
```
        assertEquals(expectedStarTreeMetadata.getDimensionFields().size(), resultStarTreeMetadata.getDimensionFields().size());
        for (int i = 0; i < expectedStarTreeMetadata.getDimensionFields().size(); i++) {
            assertEquals(expectedStarTreeMetadata.getDimensionFields().get(i), resultStarTreeMetadata.getDimensionFields().get(i));
        }
```